scripts/dotnet-cli-build/CompileTargets.cs

```diff
@@ -619,14 +619,18 @@ private static BuildTargetResult CompileCliSdk(BuildTargetContext c, DotNetCli d
 
             foreach (var project in ProjectsToPublish)
             {
+                var srcDir = Path.Combine(c.BuildContext.BuildDirectory, "src");
+                FS.CleanBinObj(c, srcDir);
+
                 dotnet.Publish(
                     "--native-subdirectory",
                     "--output",
                     outputDir,
                     "--configuration",
                     configuration,
-                    Path.Combine(c.BuildContext.BuildDirectory, "src", project))
-                    .Environment("DOTNET_BUILD_VERSION", buildVersion.VersionSuffix)
+                    "--version-suffix",
```

Comment:
I think this would be more readable if the arg name and value were on the same line:

```
"--configuration", configuration,
 "--version-suffix", buildVersion.CommitCountString,
Path.Combine(...)
```